### PR TITLE
Add: automate upgrade case OCP-23501

### PR DESCRIPTION
**Pull request by duanwei33**
@piqin 
I created a new pr with the update for your comments:
1. Add "storage" operator version matchs the current cluster version
2. Add check storage class name from the "storage" operator

[Pass log](https://privatebin-it-iso.int.open.paas.redhat.com/?0560959c181fe174#IvJRvYs7A8TaRRo0O2GCIpZXUrsezb8vTKsm3lmWZJ0=)

### CODE REVIEW

File: testdata/storage/misc/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    matchLabels:
+      app: hello-storage
+  template:
+    metadata:
+      labels:
+        app: hello-storage
+    spec:
+      containers:
+      - name: hello-storage
+        image: docker.io/aosqe/storage@sha256:a05b96d373be86f46e76817487027a7f5b8b5f87c0ac18a246b018df11529b40
```
Comment (qinpingli) [inline testdata/storage/misc/deployment.yaml:17]: use the image in quay.io?

**Comment by qinpingli** [comment]
only one comment about using test image in quay.io. Others LGTM.

File: features/upgrade/storage/upgrade.feature [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    When I run oc create over "<%= BushSlicer::HOME %>/testdata/storage/misc/deployment.yaml" replacing paths:
+      | ["metadata"]["name"]                                                             | mydep01-ocp-23501 |
+      | ["spec"]["template"]["metadata"]["labels"]["action"]                             | upgrade-prepare   |
+      | ["spec"]["template"]["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc01-ocp-23501 |
+    Then the step should succeed
+    And I wait up to 180 seconds for the steps to pass:
+    """
+    And a pod becomes ready with labels:
+      | action=upgrade-prepare |
+    """
```
Comment (liangxia) [inline features/upgrade/storage/upgrade.feature:60]: The step `pod becomes ready with` contains the wait logic, so no need to wrap it in another wait.

Comment (duanwei33) [reply features/upgrade/storage/upgrade.feature]: Thanks for pointing out, accepted.

**Pull request merged**